js/index.js

```diff
@@ -48,5 +73,46 @@ function createCardGrid() {
 function getRandomIndex() {
   return Math.floor(Math.random() * imagePickList.length);
 }
-
 createCardGrid();
+
+//how much time has passed since you first clicked on a card.
+let startTime;
+let timerInterval;
+let clickedCards = 0;
+
+function startTimer() {
+  if (!startTime) {
+    startTime = Date.now();
+    timerInterval = setInterval(updateTimer, 1000);
+  }
+
+  // Check if all cards have been clicked
+  if (clickedCards === cardCount) {
+    clearInterval(timerInterval);
+  }
+}
+
+function updateTimer() {
+  const currentTime = Date.now();
+  const timePassed = Math.floor((currentTime - startTime) / 1000);
+  timerDisplay.innerText = `Timer: ${timePassed} seconds`;
+}
+
+//Reset Button
+
+function resetGame() {
+  clearInterval(timerInterval);
+  startTime = null;
+  clickedCards = 0;
+  timerDisplay.innerText = "Timer: 0 seconds";
+  resetCards();
+}
+
+function resetCards() {
+  const items = document.querySelectorAll(".cards div");
+  items.forEach((element) => {
+    element.children[0].setAttribute("src", "./images/back-pic.jpg");
+  });
```

Comment:
Since you have this functionality to reset the game, it would be nice to shuffle the cards again, so after resetting it is a different game, right? 🙏🏽

Reply:
done